irohad/ordering/impl/ordering_service_impl.cpp

```diff
@@ -74,20 +74,21 @@ namespace iroha {
 
       // Save proposal height to the persistent storage.
       // In case of restart it reloads state.
-      persistent_state_->saveProposalHeight(proposal_height);
-
-      publishProposal(std::move(proposal));
+      if (persistent_state_->saveProposalHeight(proposal_height)) {
```

Comment:
Such implicit behaviour can lead to debugging issues. Consider another approach, such as throwing `std::runtime_error`, as it is done in Simulator, or storing a queue of proposals, which is a friendlier approach.

Reply:
Also logging negative case is definitely required.

Reply:
We are trying to avoid exceptions afair. And I do like this behavior, because there is no unpredictable unwindings
It might be a better way for changing method signature